### ISSUE TITLE
[spike] DR | Investigate 400 error

### ISSUE THREAD

**Issue by data-doge** [department-of-veterans-affairs/va.gov-team#65839]
At around 4:07PM ET on Sept. 19, 2023, we got a 400 error on the [HLR contestable issues endpoint](https://vagov.ddog-gov.com/dashboard/kiy-k3e-5ta/benefits---higher-level-reviews?refresh_mode=paused&from_ts=1695067523797&to_ts=1695067823797&live=false) and the [HLR submission endpoint](https://vagov.ddog-gov.com/dashboard/kiy-k3e-5ta/benefits---higher-level-reviews?refresh_mode=paused&from_ts=1695067523797&to_ts=1695067823797&live=false).

See [relevant time range](https://vagov.ddog-gov.com/dashboard/kiy-k3e-5ta/benefits---higher-level-reviews?refresh_mode=paused&from_ts=1695067523797&to_ts=1695067823797&live=false).

We have gotten three of these false positive errors across the HLR and SC forms over the past two days.

**Comment by data-doge** [comment]
This is a LH 504 being re-characterized by vets-api as a 400 error. Need to log the actual status code to DataDog. See http://sentry.vfs.va.gov/organizations/vsp/issues/199230/?environment=production&project=3&query=is%3Aunresolved+user.id%3Aa8d7636713484d788be158314509cbb4&statsPeriod=14d.

**Comment by data-doge** [comment]
Immediate follow-up: double-check whether we ask Veteran to re-try later on 504. Take screenshot and post here. Tag Sade and Eileen. Even a "try again" message could be unacceptably vague.

**Comment by data-doge** [comment]
We experience this same recharacterization of 504s to 400s for SC as well. See http://sentry.vfs.va.gov/organizations/vsp/issues/199517/?environment=production&project=3&query=is%3Aunresolved+user.id%3A37e08ab3243440ad955827b6a59ff1a3&statsPeriod=14d. 

For that reason, I'll update the title of this ticket to `DR | ...` rather than `HLR | ...`

**Comment by rocio-desantiago** [comment]
504 error message upon submitting

<img width="637" alt="Screenshot 2023-09-21 at 11 31 48 AM" src="https://github.com/department-of-veterans-affairs/va.gov-team/assets/37054305/69a625e9-fc7a-4949-8e0c-ccf988fc3fb4">

**Comment by rocio-desantiago** [comment]
Note for self: 
replaced line 25 with `render json: {}, status: :gateway_timeout` in `app/controllers/v1/higher_level_reviews_controller.rb`

**Comment by saderagsdale** [comment]
@rocio-desantiago @data-doge Let us know if you find a clearer recommendation that we can make to Veterans in the event of getting the error.

**Comment by rocio-desantiago** [comment]
For contestable issues (compensation) both 400 and 504 are resulting in the same error message.
[Uploading... Screenshot 2023-09-22 at 11.57.14 AM.png]

**Comment by rocio-desantiago** [comment]
@data-doge  
How do we update DD log to log actual status code (504)?

**Comment by saderagsdale** [comment]
@rocio-desantiago @data-doge any action needed on this or is it unblocked?

**Comment by kylesoskin** [comment]
@data-doge yea I belive everything you said to be true. It will be a backend change to adjust that return. This also appears to happen for 429 responses as well.

**Comment by HeatherWidmont** [comment]
@data-doge will create a new ticket for this & estimate it, then this spike ticket will be closed.

**Comment by data-doge** [comment]
Closing this spike in favor of https://github.com/department-of-veterans-affairs/va.gov-team/issues/66874
